src/fsharp/AccessibilityLogic.fs

```diff
@@ -249,12 +249,35 @@ let private IsILMethInfoAccessible g amap m adType ad ilminfo =
 let GetILAccessOfILPropInfo (ILPropInfo(tinfo, pdef)) =
     let tdef = tinfo.RawMetadata
     let ilAccess =
-        match pdef.GetMethod with 
-        | Some mref -> (resolveILMethodRef tdef mref).Access 
-        | None -> 
-            match pdef.SetMethod with 
-            | None -> ILMemberAccess.Public
-            | Some mref -> (resolveILMethodRef tdef mref).Access
+        match pdef.GetMethod, pdef.SetMethod with 
+        | Some mref, None 
+        | None, Some mref -> (resolveILMethodRef tdef mref).Access
+
+        | Some mrefGet, Some mrefSet ->
+            let getA = (resolveILMethodRef tdef mrefGet).Access
+            let setA = (resolveILMethodRef tdef mrefSet).Access
+
+            // use the accessors to determine the visibility of the property
+            match getA, setA with
```

Comment:
This looks really good. The order to determine most visible is now determined here and should give us the same effect.

Reply:
I'm in favor of this rather than having a specific order for the DU.

Reply:
My only suggestion is the follow the ordering that was dictated by Roslyn:
```
   | CompilerControlled
    | Private
    | FamilyAndAssembly // protected and internal
    | Family // protected
    | Assembly // internal
    | FamilyOrAssembly // protected or internal
    | Public
```

Reply:
I assume you are suggesting adding compiler controlled ... I can for sure do it, although for the life of me I have no idea what it even means for property accessors in F# :-)

Reply:
I'm still finding this solution has more moving pieces than the original with the exhaustive test I added that Jon asked.

In anycase, it is good to have more top level testing that are added here.

I think you know better what fits the best in the compiler maintenance mindset, main thing is to have the bug fixed in a future release :)

Maybe a comment on the match that order of cases is critical to remain "from most visible to least visible" wouldn't hurt.

Reply:
@smoothdeveloper , good call on the comment, I will add one.  The test case for the DU was certainly important if we were to rely on DU order it for this feature, however, we are not depending on that specific ordering with this change.

Reply:
added